src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,10 +1,66 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void
+};
+
+export const NewMovie = (props: Props) => {
+  const [count, setCount] = useState(0);
+  const [disabled, setDisabled] = useState(true);
+  const { onAdd } = props;
+  const [movieTitle, setMovieTitle] = useState('');
+  const [movieDescription, setMovieDescription] = useState('');
+  const [movieImageUrl, setMovieImageUrl] = useState('');
+  const [movieImdbUrl, setMovieImdbUrl] = useState('');
+  const [movieImdbId, setMovieImdbId] = useState('');
+
+  const [movie, setMovie] = useState({
+    title: movieTitle,
+    description: movieDescription,
+    imgUrl: movieImageUrl,
+    imdbUrl: movieImdbUrl,
+    imdbId: movieImdbId,
+  });
+
+  useEffect(() => {
+    const newMovie = {
+      title: movieTitle.trim(),
+      description: movieDescription.trim(),
+      imgUrl: movieImageUrl.trim(),
+      imdbUrl: movieImdbUrl.trim(),
+      imdbId: movieImdbId.trim(),
+    };
+
+    setMovie(newMovie);
+    if (newMovie.title.length !== 0
```

Comment:
Fix all cases and move to variable
```suggestion
    if (Boolean(newMovie.title.length)
```